src/casp/services/api/routers/cell_operations_router.py

```diff
@@ -11,13 +12,11 @@
 cellarium_general_service = services.CellariumGeneralService()
 
 
-@cell_operations_router.post(
-    "/annotate", response_model=t.Union[t.List[schemas.QueryCellDevDetails], t.List[schemas.QueryCell]]
-)
+@cell_operations_router.post("/annotate", response_model=schemas.QueryAnnotationCellTypeCountType)
```

Comment:
I think it is cleaner to have a single `/annotate` URL with the strategy field, rather than `/annotate` and `/annotate-ontology-aware`.

Reply:
Normally I would agree but the problem is that the parameters changes the response shape. It's generally better practice to have that in a separate api call in that case since dealing with polymorphism on the client is a pretty bad experience for users

Reply:
OK, so let's break backwards compatibility and call this `/annotate-cell-type-count-strategy`. It doesn't make sense a "default" `/annotate`, especially one that we are walking away from soon. Perhaps we could revisit the issue of having a polymorphic `/annotate`. We are likely going to study other types of query summarization strategies soon (e.g. disease, publication, etc). It is fine to have separate API calls for different strategies, but it will quickly go out of control if we have many more strategies. I am not too worried about the client-side UX. The users always get a list of dicts, and they know what kind of dict they are getting based on what strategy they have chosen...

Reply:
Hmm, not sure I agree with that.  By that description, why bother having any structure at all?  These are fundamentally different objects being returned so it seems reasonable to my to have them be different API calls.  I think returning arbitrary json with no help to the user is not a pattern I feel particularly good about promoting.

If we're going to have an arbitrary amount of these, we should think about what type of response could encapsulate all of these but honestly, having an API for each doesn't seem all that bad to me.

As for breaking backwards compatibility, I don't know that we need to do that.  We could have annotate be deprecated and then migrate 10x to the new version. Just trying not to pull the rug out from under the,/

Reply:
Also worth talking about in person...I tend to come off as a jerk when I have these conversations written down and definitely don't mean to!